etk/extractors/table_extractor.py

```diff
@@ -212,50 +213,131 @@ def extract(self, html_doc, min_data_rows = 1):
                     row_data = row_data.replace("\\t", "").replace("\\r", "").replace("\\n", "")
-                    if row_data != '':
-                        row_len_list.append(len(row_data))
-                        row_tdcount = len(soup_row.findAll('td')) + len(soup_row.findAll('th'))
-                        if row_tdcount > max_tdcount:
-                            max_tdcount = row_tdcount
-                        tdcount += row_tdcount
-                        img_count += len(soup_row.findAll('img'))
-                        href_count += len(soup_row.findAll('a'))
-                        inp_count += len(soup_row.findAll('input'))
-                        sel_count += len(soup_row.findAll('select'))
-                        colspan_count += row_data.count("colspan")
-                        colon_count += row_data.count(":")
-                        len_row += 1
-                        table_data += row
-                        # row_dict["row"] = str(row)
-                        cell_list = list()
-                        for index_col, td in enumerate(soup_row.findAll('th')):
-                            cell_dict = dict()
-                            cell_dict["cell"] = str(td)
-                            # cell_dict["text"] = [{"result": {"value": ''.join(td.stripped_strings)}}]
-                            cell_dict["text"] = ' '.join(td.stripped_strings)
-                            cell_dict["id"] = 'row_{0}_col_{1}'.format(index_row, index_col)
-                            avg_cell_len += len(cell_dict["text"])
-                            cell_list.append(cell_dict)
-                        for index_col, td in enumerate(soup_row.findAll('td')):
-                            cell_dict = dict()
-                            cell_dict["cell"] = str(td)
-                            # cell_dict["text"] = [{"result": {"value": ''.join(td.stripped_strings)}}]
-                            cell_dict["text"] = ' '.join(td.stripped_strings)
-                            cell_dict["id"] = 'row_{0}_col_{1}'.format(index_row, index_col)
-                            avg_cell_len += len(cell_dict["text"])
-                            cell_list.append(cell_dict)
-                        avg_row_len_dev += TableExtraction.pstdev([len(x["text"]) for x in cell_list])
-                        row_dict["cells"] = cell_list
-                        row_dict["text"] = self.row_to_text(cell_list)
-                        row_dict["html"] = self.row_to_html(cell_list)
-                        row_dict["id"] = "row_{}".format(index_row)
-                        row_list.append(row_dict)
-
+                    
+                    row_len_list.append(len(row_data))
+                    row_tdcount = len(row.findAll('td')) + len(row.findAll('th'))
+                    if row_tdcount > max_tdcount:
+                        max_tdcount = row_tdcount
+                    tdcount += row_tdcount
+                    img_count += len(row.findAll('img'))
```

Comment:
It may be cleaner to do:

    counts["img"] += ...
    counts["href"] += ...

than having a separate variable for each.